src/resources/Resource.ts

```diff
@@ -58,12 +58,23 @@ type ObjectType =
     | "object"
     | "string"
     | "symbol"
-    | "undefined";
+    | "undefined"
+    | "blob";
 
-const getObjectType = (o: unknown): ObjectType => {
-    if (o === null) return "null";
-    if (Array.isArray(o)) return "array";
-    return typeof o;
+const getObjectType = (data: unknown): ObjectType => {
```

Comment:
This function was added to destructure form data in order to prevent adding the multipart data for JSON. We just need to be sure we do not destructure the blobs as well to prevent the loss of the important class properties.